test/gtest/ucm/malloc_hook.cc

```diff
@@ -241,8 +249,10 @@ void test_thread::test() {
     /* Release old pointers (should not crash) */
     old_ptrs.clear();
 
+    pthread_mutex_lock(&m_stats_lock);
```

Comment:
Then current solution just hides a real bug. Pls add WIP label. I'll add a barrier, one more check to non zero length in unmap event handler and continue debugging.